src/agent/border_agent.cpp

```diff
@@ -250,61 +288,74 @@ void BorderAgent::StartPublishService(void)
     {
         mPublisher->Start();
     }
+
+exit:
+    return;
 }
 
 void BorderAgent::StopPublishService(void)
 {
-    otbrLog(OTBR_LOG_INFO, "Stop publishing service");
+    VerifyOrExit(mPublisher != NULL);
```

Comment:
Either `avahi` or `mDNSResponder` would be used, `mPublisher` should not be `NULL` during running, right?